docs/gen_reference_doc.py

```diff
@@ -729,266 +824,311 @@ def consume_ifdef(lno, lines, warn_on_ifdefs = False):
+    global link_targets
+    link_targets.append(target)
+    return "`%s`__" % name
+
+
+def dump_link_targets(indent=''):
+    global link_targets
+    ret = '\n'
+    for l in link_targets:
+        ret += '%s__ %s\n' % (indent, l)
+    link_targets = []
+    return ret
+
+
+def heading(string, c, indent=''):
+    string = string.strip()
+    return '\n' + indent + string + '\n' + indent + (c * len(string)) + '\n'
+
 
 def render_enums(out, enums, print_declared_reference, header_level):
-	for e in enums:
-		print >>out, '.. raw:: html\n'
-		print >>out, '\t<a name="%s"></a>' % e['name']
-		print >>out, ''
-		print >>out, heading('enum %s' % e['name'], header_level)
-
-		print_declared_in(out, e)
-
-		width = [len('name'), len('value'), len('description')]
-
-		for i in range(len(e['values'])):
-			e['values'][i]['desc'] = linkify_symbols(e['values'][i]['desc'])
-
-		for v in e['values']:
-			width[0] = max(width[0], len(v['name']))
-			width[1] = max(width[1], len(v['val']))
-			for d in v['desc'].split('\n'):
-				width[2] = max(width[2], len(d))
-
-		print >>out, '+-' + ('-' * width[0]) + '-+-' + ('-' * width[1]) + '-+-' + ('-' * width[2]) + '-+'
-		print >>out, '| ' + 'name'.ljust(width[0]) + ' | '  + 'value'.ljust(width[1]) + ' | ' + 'description'.ljust(width[2]) + ' |'
-		print >>out, '+=' + ('=' * width[0]) + '=+=' + ('=' * width[1]) + '=+=' + ('=' * width[2]) + '=+'
-		for v in e['values']:
-			d = v['desc'].split('\n')
-			if len(d) == 0: d = ['']
-			print >>out, '| ' + v['name'].ljust(width[0]) + ' | '  + v['val'].ljust(width[1]) + ' | ' + d[0].ljust(width[2]) + ' |'
-			for s in d[1:]:
-				print >>out, '| ' + (' ' * width[0]) + ' | '  + (' ' * width[1]) + ' | ' + s.ljust(width[2]) + ' |'
-			print >>out, '+-' + ('-' * width[0]) + '-+-' + ('-' * width[1]) + '-+-' + ('-' * width[2]) + '-+'
-		print >>out, ''
-
-		print >>out, dump_link_targets()
+    for e in enums:
+        print('.. raw:: html\n', file=out)
+        print('\t<a name="%s"></a>' % e['name'], file=out)
+        print('', file=out)
+        print(heading('enum %s' % e['name'], header_level), file=out)
+
+        print_declared_in(out, e)
+
+        width = [len('name'), len('value'), len('description')]
+
+        for i in range(len(e['values'])):
+            e['values'][i]['desc'] = linkify_symbols(e['values'][i]['desc'])
+
+        for v in e['values']:
+            width[0] = max(width[0], len(v['name']))
+            width[1] = max(width[1], len(v['val']))
+            for d in v['desc'].split('\n'):
+                width[2] = max(width[2], len(d))
+
+        print('+-' + ('-' * width[0]) + '-+-' + ('-' * width[1]) + '-+-' + ('-' * width[2]) + '-+', file=out)
+        print(
+            '| ' +
+            'name'.ljust(
+                width[0]) +
+            ' | ' +
+            'value'.ljust(
+                width[1]) +
+            ' | ' +
+            'description'.ljust(
+                width[2]) +
+            ' |',
+            file=out)
```

Comment:
this line seems to have been chopped up in quite small pieces. was this deliberate?

Reply:
Nope, looks like autopep8 went a bit wild with this. It was probably trying to deal with the max-line-length.  I'll clean this up.